it/src/main/java/org/corfudb/universe/node/docker/CorfuServerDockerized.java

```diff
@@ -0,0 +1,214 @@
+     *
+     * @return docker container id
+     */
+    private String deployContainer() {
+        ContainerConfig containerConfig = buildContainerConfig();
+
+        String id;
+        try {
+            ContainerCreation container = docker.createContainer(containerConfig, params.getName());
+            id = container.id();
+
+            addShutdownHook();
+
+            docker.connectToNetwork(id, docker.inspectNetwork(universeParams.getNetworkName()).id());
+
+            docker.startContainer(id);
+        } catch (InterruptedException | DockerException e) {
+            Thread.currentThread().interrupt();
+            throw new NodeException("Can't start a container", e);
+        }
+
+        return id;
+    }
+
+    private ContainerConfig buildContainerConfig() {
+        // Bind ports
+        String[] ports = {String.valueOf(params.getPort())};
+        Map<String, List<PortBinding>> portBindings = new HashMap<>();
+        for (String port : ports) {
+            List<PortBinding> hostPorts = new ArrayList<>();
+            hostPorts.add(PortBinding.of(ALL_NETWORK_INTERFACES, port));
+            portBindings.put(port, hostPorts);
+        }
+
+        HostConfig hostConfig = HostConfig.builder()
+                .portBindings(portBindings)
+                .autoRemove(true)
+                .build();
+
+        // Compose command line for starting Corfu
+        String cmdLine = String.format(
+                "java -cp *.jar org.corfudb.infrastructure.CorfuServer %s",
+                getCommandLineParams()
+        );
+
+        return ContainerConfig.builder()
+                .hostConfig(hostConfig)
+                .image(IMAGE_NAME)
+                .hostname(params.getName())
+                .exposedPorts(ports)
+                .cmd("sh", "-c", cmdLine)
+                .build();
+    }
+
+    private void addShutdownHook() {
+        // Just in case a test failed and didn't kill the container
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                kill();
+            } catch (Exception e) {
+                log.debug("Corfu server shutdown hook. Can't kill container: {}", params.getName());
+            }
+        }));
+    }
+
+    private static String getAppVersion() {
+        MavenXpp3Reader reader = new MavenXpp3Reader();
+        Model model;
+        try {
+            model = reader.read(new FileReader("pom.xml"));
+            return model.getParent().getVersion();
+        } catch (IOException | XmlPullParserException e) {
+            throw new NodeException("Can't parse application version", e);
+        }
+    }
+
+    /**
+     * This method create a command line string for starting Corfu server
+     *
+     * @return command line parameters
+     */
+    private String getCommandLineParams() {
+        StringBuilder cmd = new StringBuilder();
+        cmd.append("-a ").append(ALL_NETWORK_INTERFACES);
```

Comment:
If you are using -a, then it should be the actual address of the node, otherwise the fault detectors wont work.

Reply:
Looks like this is a major limitation which is not easy to solve. 
There are 2 good enough ways to solve it:
 - use corfu client proxy instead of local corfu runtime. I like this solution because it is similar to the real case, this is exactly how we use corfu.
 - run unit tests in a docker container too. This approach could work but it takes time to implement it and there are some limitations.

So, Wenbin work on a corfu proxy, let's use it when he finishes